netket/experimental/operator/_fermion_operator_2nd_base.py

```diff
@@ -539,7 +368,7 @@ def _op__imatmul__(self, other):
         return self
 
     def _op__matmul__(self, other):
-        if not isinstance(other, FermionOperator2nd):
+        if not isinstance(other, FermionOperator2ndBase):
```

Comment:
```suggestion
        if not isinstance(other, type(self)):
```
Do we really want to allow adding numba + jax ?

(same for for add below)

Reply:
I think it's ok for now. At least until we allow for manipulating them inside jit expressions (which we don't yet) this should not lead to any problems...